torch/csrc/jit/register_prim_ops.cpp

```diff
@@ -757,19 +769,20 @@ RegisterOperators reg({
         [](const Node* node) -> Operation {
           const auto num_inputs = node->inputs().size();
           ListTypePtr lt = node->output()->type()->expect<ListType>();
-          if(IntType::get() == lt->getElementType()) {
+          if (IntType::get() == lt->getElementType()) {
             return listConstruct<int64_t>(num_inputs);
-          } else if(FloatType::get() == lt->getElementType()) {
+          } else if (FloatType::get() == lt->getElementType()) {
             return listConstruct<double>(num_inputs);
           } else if (lt->getElementType() == BoolType::get()) {
             return listConstruct<bool>(num_inputs);
-          } else if (lt->getElementType()->isSubtypeOf(DynamicType::get())) {
+          } else if (lt->getElementType()->isSubtypeOf(
```

Comment:
this is the change that when we run list construct, we also consider the case that there're optional tensors, and run it with call to `toOptionalTensor`

Reply:
Huh? Why did this code get changed? A list of optional tensors is not the same thing as a list of tensors, and should _not_ be going down this pathway.

Reply:
Yes list of optional tensors are not list of tensors in jit, what I did here is that ListConstruct in the interpreter could accept both `List[Optional[Tensor]]` or `List[Tensor]`, and `List[Optional[Tensor]]` will need to be translate to `TensorList` as well during execution to be able to pass to ATen functions(the two types are equivalent in C++ still).